packages/widget-v2/src/components/MainButton.tsx

```diff
@@ -0,0 +1,168 @@
+import { styled } from 'styled-components';
+import { Row } from './Layout';
+import { SmallText, Text } from './Typography';
+import { PlusIcon } from '../icons/PlusIcon';
+import { useTheme } from 'styled-components';
+import { CheckmarkIcon } from '../icons/CheckmarkIcon';
+import { LeftArrowIcon, RightArrowIcon } from '../icons/ArrowIcon';
+import { SwapIcon } from '../icons/SwapIcon';
+import { WarningIcon } from '../icons/WarningIcon';
+
+export enum ICONS {
+  plus,
+  checkmark,
+  rightArrow,
+  leftArrow,
+  swap,
+  warning,
+}
+
+const iconMap = {
+  [ICONS.plus]: PlusIcon,
+  [ICONS.checkmark]: CheckmarkIcon,
+  [ICONS.rightArrow]: RightArrowIcon,
+  [ICONS.leftArrow]: LeftArrowIcon,
+  [ICONS.swap]: SwapIcon,
+  [ICONS.warning]: WarningIcon,
+  none: () => null,
+};
+
+export type MainButtonProps = {
+  label: string;
+  icon?: ICONS | 'none';
+  leftIcon?: ICONS | 'none';
+  backgroundColor?: string;
+  disabled?: boolean;
+  loading?: boolean;
+  loadingTimeString?: string;
+  onClick?: () => void;
+};
+
+export const MainButton = ({
+  label,
+  icon = 'none',
+  leftIcon = 'none',
+  backgroundColor,
+  disabled,
+  loading,
+  loadingTimeString,
+  onClick,
+}: MainButtonProps) => {
+  const theme = useTheme();
+  backgroundColor ??= disabled ? theme.secondary.background : theme.brandColor;
+
+  const Icon = iconMap[icon];
+  const LeftIcon = iconMap[leftIcon];
+
+  if (loading) {
+    return (
+      <StyledMainButton
+        align="center"
+        justify="center"
+        backgroundColor={theme.backgroundColor}
+        loading={loading}
+      >
```

Comment:
I didn't add onClick handler on this because it seems like the "loading" buttons shouldn't be clickable

Reply:
yeah onClick can be undefined so lgtm